src/main/java/alice/parser/DateParser.java

```diff
@@ -0,0 +1,45 @@
+package alice.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/** Handles datetime and string conversions. */
+public class DateParser {
+    private static final DateTimeFormatter[] DATETIME_FORMATTERS = new DateTimeFormatter[]{
+        DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"),
+        DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"),
+        DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"),
+        DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm"),
+    };
+
+    /**
+     * Parses a datetime in yyyy-MM-dd HH:mm, yyyy-MM-dd HHmm,
+     * dd-MM-yyyy HH:mm or dd-MM-yyyy HHmm format.
+     * 
+     * @param  dateString             the date string to parse
+     * @throws DateTimeParseException if the date string is not well=formed
+     * @return                        the parsed LocalDateTime
+     */
+    public static LocalDateTime parseDateString(String dateString) throws DateTimeParseException {
+        for (DateTimeFormatter formatter :  DATETIME_FORMATTERS) {
+            try {
+                return LocalDateTime.parse(dateString, formatter);
+            } catch (DateTimeParseException exception) {
+                // try another format
```

Comment:
perhaps print something, even if you're not doing anything with the exception object caught